packages/ckeditor5-paragraph/src/paragraphcommand.ts

```diff
@@ -52,7 +59,14 @@ export default class ParagraphCommand extends Command {
 		const document = model.document;
 
 		model.change( writer => {
-			const blocks = ( options.selection || document.selection ).getSelectedBlocks();
+			const selection = options.selection || document.selection;
+
+			// Don't execute command if selection is in non-editable place.
+			if ( !model.isSelectableEditable( selection ) ) {
+				return;
+			}
```

Comment:
These lines (62-67) should be outside of `model.change()` so that it is not called if not needed.